### DISCUSSION TITLE
MAX7219 Feature: DisplayScrollText repeat the text

### DISCUSSION THREAD

**Discussion by freifunkerffm** [arendst/Tasmota#13075] [Q&A] [answered]
Hi folks,

I have played around with MAX7219 (controller to support 8 times 7-Segement by 3 MAX7218 wires, VCC and GND).
The latest TASMOTA9.5.0.8 is supporting this controller and display.
I got it to work to display text, what I did not got working properly is the command:
DISPLAYSCROLLTEXT
It  is scrolling the text exactly one time!
If I set a simple DISPLAYTEXT 12345 this will be displayed nearly forever (until power off or the next DISPLAYTEXT is performed).

IMHO I would await that DISPLAYSCROLLTEXT is scrolling the text over and over until the next DISPLAYSCROLLTEXT with a different text is arriving.

Now the question is that a "bug" or a feature?
May there is another command that makes the text appear again. 
Or is there a "workarround" known/available?
I would like to avoid to create a "Loop Monster" in Node-Red to resend the scroll text every 30 seconds, that is also a bit against MQTT in my eyes.

Any hints for solving that?

**Comment by barbudor** [comment]
What have you set as `DisplayScrollDelay` ?

**Reply by freifunkerffm** [reply]
I set it to 15,  that makes the speed of scrolling as slow as possible, setting it lower makes the scroll faster. I also thought that this is may the time between the resend but I was wrong.
I waited a longer time but the display keeps dark after the first scroll

**Reply by barbudor** [reply]
Ok, I misunderstood. It scrolls once but only once.
Let me check

**Reply by barbudor** [reply]
From the code, it is designed to scroll only once. 
May be the author may add an option for single or continuous scroll

@ajithvasudevan Are you the author ? Can you help ?

**Reply by ajithvasudevan** [reply]
> @ajithvasudevan Are you the author ? Can you help ?

Yes, I am, and I can add functionality to make it scroll  "n" times and stop, or scroll "indefinitely".

Is it okay if the default command  **DISPLAYSCROLLTEXT  &lt;Text&gt;**  scrolls indefinitely, and the command **DISPLAYSCROLLTEXT  n  &lt;Text&gt;**  scrolls "n" times and stops?
 
Or, would you prefer separate commands to scroll "n" times as well as to scroll indefinitely?

**Comment by freifunkerffm** [comment]
@ajithvasudevan 
I would be fine with your suggestion that DISPLAYSCROLLTEXT <Text> scrolls indefinitely, and the command DISPLAYSCROLLTEXT n <Text> scrolls "n" times and stops.

Thanks a lot

Kai from Germany

**Reply by barbudor** [reply]
@ajithvasudevan 

I'm not sure if it is easy for you to distinguish between `DisplayScrollText <n> <text>` and `DisplayScrollText <text>` if `<text>` can include spaces.
You may use the  Mailbox.index so the commands would be 
`DisplayScrollText <text>` or  `DisplayScrollText1 <text>`  (index==1) as today, single scroll
`DisplayScrollText2 <n> <text>` (index==2) for the new syntax

**Reply by ajithvasudevan** [reply]
> @ajithvasudevan
> 
> I'm not sure if it is easy for you to distinguish between `DisplayScrollText <n> <text>` and `DisplayScrollText <text>` if `<text>` can include spaces.
> You may use the Mailbox.index so the commands would be
> `DisplayScrollText <text>` or `DisplayScrollText1 <text>` (index==1) as today, single scroll
> `DisplayScrollText2 <n> <text>` (index==2) for the new syntax

I realized this as soon as I started thinking about it :)
In line with the rest of the 7-segment command syntax, I think **DisplayScrollText &lt;Text&gt; , &lt;n&gt;**  with   **&lt;n&gt;** being optional, should be fine.

**Comment by ajithvasudevan** [comment]


**Reply by ajithvasudevan** [reply]
Sorry, I accidentally deleted the previous message, which I am copying below:

> I have submitted a PR for this feature:  https://github.com/arendst/Tasmota/pull/13077

**Comment by freifunkerffm** [comment]
@ajithvasudevan: I have patched your changes back in my actual TASMOTA 9.5.0.8 (replaced the xdsp_15_tm1637.ino with your version from pull request and rebuild TASMOTA)  and DISPLAYSCROLLTEXT2 is working perfect.
Will the documentation be automatically updated?

Kai

**Reply by barbudor** [reply]
> Will the documentation be automatically updated?

Nope.
I will do it unless @ajithvasudevan prefers to do it himself
Note that the doc update will go into the Development branch and will not appear on the Tasmota web site until the next official release (there isn't a Development Doc website for now)

**Reply by ajithvasudevan** [reply]
I will do the documentation update, if you don't mind. I may have a few minor corrections to the existing documentation as well.

**Reply by barbudor** [reply]
All good, always better when it comes from the author 👍

**Comment by ajithvasudevan** [comment, marked as answer]
And here is the PR for the updated documentation:  https://github.com/tasmota/docs/pull/800

@freifunkerffm, as @barbudor mentioned, it will take a while for this to become available on the documentation website.

But if you're in a hurry, you could clone https://github.com/ajithvasudevan/docs/tree/max7912_scrolltext_enhancement and run the `mkdocs serve` command from  within the cloned folder to have your own local server serving the updated documentation.

Note:  To install `mkdocs`, see here: https://www.mkdocs.org/getting-started/
